src/auto_contiguous.cpp

```diff
@@ -64,7 +64,10 @@ void auto_contiguous::apply(module& m) const
         if(ins->outputs().empty() and ins != last)
             continue;
         shape s = ins->get_shape();
-        if(not s.dynamic() and not s.standard() and s.elements() != 0)
+        // If s is not standard layout or has out of sequence strides, insert "contiguous" op
+        // to make a standard shape
+        if(not s.dynamic() and (not s.transposed() or s.normalize_standard() != s) and
```

Comment:
Remove `not s.transposed()` because we want to insert contiguous when its transposed. 

Also to fix the tests we need to skip contiguous for single scalars, so we can change the condition from `s.elements() != 0` to `s.elements() > 1`, that should fix the issue.

So the final condition should look like: `if(not s.dynamic() and s.as_standard() != s and s.elements() > 1)`.

Reply:
Tried it, but I'm still seeing multiple test failures with the same message, for instance:
```
[   RUN    ] test_select_module_conv
test_verify: /workspace/AMDMIGraphX/src/shape.cpp:57: migraphx::shape_impl::shape_impl(shape::type_t, std::vector<std::size_t>): Assertion `t != shape::tuple_type' failed.
```
Not sure what we'd change to fix that.

Reply:
Update: Also a few tests failed with this message:
```
 34/357 Test #351: test_api_gpu ..............................................................Subprocess aborted***Exception:   0.49 sec
[   RUN    ] load_and_run
MIGraphX Error: /workspace/AMDMIGraphX/src/onnx/onnx_parser.cpp:407: parse_graph: PARSE_GRAPH: invalid onnx file. Input "0" is unavailable due to unordered nodes!
terminate called after throwing an instance of 'std::runtime_error'
  what():  Failed to call function
```
Can an `onnx_parser` error be caused by a bug in an `auto_contiguous` pass?